.azurepipelines/GetCargoMake.yml

```diff
@@ -0,0 +1,114 @@
+## @file
+# Azure Pipeline to download Cargo Make and save it as a pipeline artifact that
+# can be accessed by other pipelines.
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+schedules:
+# At 1:00 on Monday
+# https://crontab.guru/#0_1_*_*_1
+- cron: 0 1 * * 1
+  branches:
+    include:
+    - main
+  always: true
+
+jobs:
+- job: Update_Cargo_Make
+  displayName: Update Cargo Make
+
+  pool:
+    vmImage: windows-latest
+
+  steps:
+  - checkout: none
+    clean: true
+    fetchTags: false
+    persistCredentials: True
+
+  - script: pip install requests --upgrade
+    displayName: Install and Upgrade pip Modules
+    condition: succeeded()
+
+  - task: PythonScript@0
+    displayName: Download and Stage Cargo Make
+    env:
+      BINARIES_DIR: "$(Build.BinariesDirectory)"
+      DOWNLOAD_DIR: "$(Build.ArtifactStagingDirectory)"
+    inputs:
+      scriptSource: inline
+      script: |
+          import os
+          import requests
+          import shutil
+          import tarfile
+          import zipfile
+          from pathlib import Path
+          from typing import Iterable
+
+          REPO_URL = "https://api.github.com/repos/sagiegurari/cargo-make/releases"
+          BINARIES_DIR = Path(os.environ["BINARIES_DIR"])
+          DOWNLOAD_DIR = Path(os.environ["DOWNLOAD_DIR"], "archives")
+
+          # Ensure the directories exist
+          BINARIES_DIR.mkdir(parents=True, exist_ok=True)
+          DOWNLOAD_DIR.mkdir(parents=True, exist_ok=True)
+
+          # Fetch the list of assets from the GitHub releases
+          response = requests.get(REPO_URL)
+          response.raise_for_status()
+          releases = response.json()
+
+          if len(releases) == 0:
+              print("Failed to find releases.")
+              exit(1)
+
+          # Download assets
+          for asset in releases[0]['assets']:
+              name = asset['name'].lower()
+              if ("x86_64-pc-windows-msvc" in name or "x86_64-unknown-linux-gnu" in name
+                and asset['name'].endswith(('.zip', '.tar.gz'))):
+                  filepath = DOWNLOAD_DIR / asset['name']
+                  print(f"Downloading {asset['name']}...")
+                  with requests.get(asset['browser_download_url'], stream=True) as r:
+                      with filepath.open('wb') as f:
+                          for chunk in r.iter_content(chunk_size=8192):
+                              f.write(chunk)
+
+          # Extract files
+          for filename in DOWNLOAD_DIR.iterdir():
+              extracted_dir = DOWNLOAD_DIR / filename.stem
+
+              print(f"Extracting {filename.name}...")
+              if filename.name.endswith('.zip'):

```

Comment:
Another option to this is to grab the file extension with Path().suffixes and pass that to `shutil.unpack_archive` as 'zip' or 'gztar' and unzip it that way. Not necessary, and not sure if worth the change in code. But its an option.

``` python
shutil.unpack_archive(filename, extracted_dir, "".join(filename.suffixes[::-1]))
```

Reply:
I'll likely leave it as-is, but I'll consider it while I'm consolidating into the script file.

Reply:
Decided to keep as is given there's no substantial benefit outlined for the change.